### DISCUSSION TITLE
Sensors Unavailable: Root cause help...

### DISCUSSION THREAD

**Discussion by smoketabs** [nathanmarlor/foxess_modbus#441] [Q&A] [answered]
Hello

I'm using this integration and from time to time, I've noticed that all the sensors become unavailable.

It started to happen when I upgraded my HA instance to the latest release but also saw that they came back. So I thought nothing of it (really).  Then over the past week it's become more prominent - leading to yesterday (when I was away at work) no sensor recording from 0940 yesterday (19-oct) and until I restarted this morning.

And whilst typing this, I notice it's just become unavailable again.

Would welcome any thoughts on why this observation might be occurring?

Thanks in advance.

Connecting to HV via Modbus RS485
LAN connection to Fox Cloud.

Master : 1.60
Slave : 1.02
Manager : 1.63

Home Assistant 2023.10.3
Supervisor 2023.10.0
Operating System 11.0
Frontend 20231005.0 - latest

![Screenshot 2023-10-20 at 07 20 29](https://github.com/nathanmarlor/foxess_modbus/assets/15179147/61a5c637-0908-4db0-8b6a-a512be54b82e)
![Screenshot 2023-10-20 at 07 21 24](https://github.com/nathanmarlor/foxess_modbus/assets/15179147/0efcbd3c-31dc-4f85-9f6e-3519edbb1c56)
![Screenshot 2023-10-20 at 07 21 43](https://github.com/nathanmarlor/foxess_modbus/assets/15179147/feb67120-3be0-4077-8bcc-61f18e53cfbc)

**Comment by nathanmarlor** [comment]
Can you include some logs please?

**Reply by smoketabs** [reply]
Hi.

Thanks for getting back to me.

Here's an extract from a HA Core search for FoxESS:

Hope these help.

Thanks again.

[home-assistant_2023-10-20T07-48-26.072Z.log](https://github.com/nathanmarlor/foxess_modbus/files/13051601/home-assistant_2023-10-20T07-48-26.072Z.log)

![Screenshot 2023-10-20 at 08 49 50](https://github.com/nathanmarlor/foxess_modbus/assets/15179147/acb2be4f-344a-46d6-a03c-ad821bdcc2a8)

And same again but for modbus:

[home-assistant_2023-10-20T07-50-25.189Z.log](https://github.com/nathanmarlor/foxess_modbus/files/13051624/home-assistant_2023-10-20T07-50-25.189Z.log)

![Screenshot 2023-10-20 at 08 51 04](https://github.com/nathanmarlor/foxess_modbus/assets/15179147/f10f61d4-5550-41c4-a606-1c1fd3b4e14d)

**Reply by nathanmarlor** [reply]
Thanks, looks like your USB adapter disconnected?

**Reply by canton7** [reply]
It also looks like you have foxess_modbus, foxess_em and possibly also the native modbus integration all running. Are they all trying to use the same USB device?

**Reply by smoketabs** [reply]
I thought that too. Have resat the dongle in my NUC (oooh Matron!). Let's see if that does anything. Or perhaps my dongle is starting to fail (happens with old age...)

**Comment by smoketabs** [comment]
> It also looks like you have foxess_modbus, foxess_em and possibly also the native modbus integration all running. Are they all trying to use the same USB device?

Yeap. I have Eastron EMs spread across my consumer unit all connected up for RS485. And then your integrations.

**Reply by canton7** [reply]
Right. USB devices can't typically be opened by more than one thing, especially things which provide serial ports. And if they could, then two integrations could decide to send some bytes at the same time and they've get jumbled together, or some bytes would be received and the usb device wouldn't know which integration to give them to.

There are (eventual) plans to let foxess_modbus and foxess_em use the same USB device, but that's not going to work for the native modbus integration.

Your best bet is probably to get a tcp to modbus adapter (not tcp to RS485) which can handle multiple simultaneous connections (I think all of the ones listed on the wiki can, except the W610).

**Reply by smoketabs** [reply]
Thanks.

I'm using the Waveshare USB to RS485 (B) since day 1 and it's been faultless.

I also have Waveshare RS485 to ETH (B) PoE in my chain but was never able to get it to work - hence using the USB adaptor.

I use Mac so all that BinDoze stuff is a waste of time for me.

I'll have another go when I get time.

Thanks

**Comment by smoketabs** [comment]
In the config of the Waveshare, I always seem to have config problems via teh web interface. The guides then start to use a Windows app for config and the likes. But I can’t as I don’t have a MS device anywhere.

**Comment by smoketabs** [comment, marked as answer]
> Right. USB devices can't typically be opened by more than one thing, especially things which provide serial ports. And if they could, then two integrations could decide to send some bytes at the same time and they've get jumbled together, or some bytes would be received and the usb device wouldn't know which integration to give them to.
> 
> There are (eventual) plans to let foxess_modbus and foxess_em use the same USB device, but that's not going to work for the native modbus integration.
> 
> Your best bet is probably to get a tcp to modbus adapter (not tcp to RS485) which can handle multiple simultaneous connections (I think all of the ones listed on the wiki can, except the W610).

Hi.

Quick update.

For my use case, I seem to have now landed on a working solution.

I'm running the Waveshare RS485 to ETH (B) PoE for H1 modbus only, and have configured FoxESS Modbus & EM HACS integrations with that.

I'm also running a Waveshare USB to RS485 (B) for all of my other modbus needs (in my case, a load of Eastron EMs) and using a ModbusUSB.yaml in my config.

All seems to be stable, and previously experienced sensor unavailability has gone.

Thanks.

**Reply by canton7** [reply]
Thanks for the update! Yep that sounds like it should work fine. Yl
